### ISSUE TITLE
Incorrect log rotation

### ISSUE THREAD

**Issue by adasari** [ava-labs/avalanchego#773]
Hi,

The current log rotation is based on size and interval. Log rotation by interval is not working as expected.

At some point the log files created by node are -

```
├── 0.log
├── chain
│   ├── C
│   │   ├── 0.log
│   │   └── http
│   │       └── 0.log
│   ├── P
│   │   ├── 0.log
│   │   └── http
│   │       └── 0.log
│   └── X
│       ├── 0.log
│       ├── 1.log
│       └── http
│           └── 0.log
└── http
    └── 0.log
```

after few rotations (changed the rotation interval to 30 sec, still file size is default 8 MB) -

```
├── 0.log
├── chain
│   ├── C
│   │   ├── 0.log
│   │   ├── 1.log
│   │   ├── 2.log
│   │   ├── 3.log
│   │   └── http
│   │       └── 0.log
│   ├── P
│   │   ├── 0.log
│   │   └── http
│   │       └── 0.log
│   └── X
│       ├── 0.log
│       ├── 1.log
│       ├── 2.log
│       ├── 3.log
│       └── http
│           └── 0.log
└── http
    └── 0.log
```

When C chain have log file index 0, P chain also have log file index 0. But by the time (after two rotations ) C chain have 3 file indexes (not by exceed log file size), P chain still have only 0 index which is incorrect. The same issue with other http logs as well. Considering the rotation interval, it should have minimum 0 and 1 file index files (worst case edge scenarios)

**Comment by danlaine** [comment]
Hi, sorry for the delay in responding to this. We're probably going to include #774 in an upcoming release, which will change the logging/rotation behavior.

**Comment by danlaine** [comment]
Resolved by #774
